examples/basics/src/lib.rs

```diff
@@ -0,0 +1,104 @@
+#[macro_use]
+extern crate neon;
+
+// thanks to @rm-rf-etc for these: https://github.com/neon-bindings/neon/issues/260
+use neon::vm::{Call, JsResult};
+use neon::js::{
+  Object, // <- https://github.com/neon-bindings/neon/issues/57
+  JsFunction,
+  JsObject,
+  JsArray,
+  JsBoolean,
+  JsInteger,
+  JsNull,
+  JsNumber,
+  JsString,
+  JsUndefined};
+
+/*
+	// If you need to inspect the value of an expression:
+	let () = JsBoolean::new();
+*/
+
+// Use _ to squelch warnings, or any name starting with _
+fn get_null_sync(_: Call) -> JsResult<JsNull> {
+  Ok(JsNull::new())
+}
+
+// Use _ to squelch warnings, or any name starting with _
+fn get_undefined_sync(_: Call) -> JsResult<JsUndefined> {
+  Ok(JsUndefined::new())
+}
+
+fn get_number_sync(call: Call) -> JsResult<JsNumber> {
+  Ok(JsNumber::new(call.scope, 5f64))
+}
+
+fn get_integer_sync(call: Call) -> JsResult<JsInteger> {
+  Ok(JsInteger::new(call.scope, 5i32))
+}
+
+fn get_string_sync(call: Call) -> JsResult<JsString> {
+  Ok(JsString::new(call.scope, "my string").unwrap())
+}
+
+fn get_boolean_sync(call: Call) -> JsResult<JsBoolean> {
+  Ok(JsBoolean::new(call.scope, false))
+}
+
+fn get_array_sync(call: Call) -> JsResult<JsArray> {
+  let scope = call.scope;
+  let array = JsArray::new(scope, 3);
+
+  try!(array.set(0, JsInteger::new(scope, 1)));
```

Comment:
`try!` is kind of "old-fashioned Rust" style, since the `?` syntax was introduced in [Rust 1.13](https://blog.rust-lang.org/2016/11/10/Rust-1.13.html). The more idiomatic way to write these lines would be:

```rust
array.set(0, JsInteger::new(scope, 1))?;
array.set(1, JsInteger::new(scope, 2))?;
array.set(2, JsInteger::new(scope, 3))?;
```